WI #643: Specification Change Guidelines (#684)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/metadata/schema/column_definition/previouscolumnname.md

Reviewer: The previous column name may not be a single column. For example, in FOCUS 1.2, we proposed to incorporate AWS Cost Category into the Tag column. So the Tag column in FOCUS 1.2 is a merge of the FOCUS 1.0 Tag column and FOCUS 1.0 x_Cost_Category column. How should we represent this in the PreviousColumnName given it only allows a single string?

Reviewer 2: so this is a deprecation of the x_Cost_Category, but an addition to the data stored in the Tags column. So this is a schema change and a data implementation change at the same time. Open to ideas on what you recommend here.

Reviewer: The previous column needs to be a list, not a single value

Author: Previous column is only for rename not deprecation.

# PreviousColumnName

The PreviousColumnName field indicates that on that schema the column where the key is included was renamed.

> In cases where the PreviousColumnName is present, the following applies:

Reviewer 3: nit: We should align this with the new standard for normative requirements

Reviewer 4: We will tackle this in AI #849 in the next release.  For now, I would only ask for basic grammar cleanup, which I will recommend in a separate comment.

* The PreviousColumnName MUST not be Null

> * The PreviousColumnNameof must by of type string and be the name use in previous versions of the schema. 

Reviewer 4: ```suggestion
* The PreviousColumnName MUST be of type String and be the name use in previous versions of the schema. 
```

Reviewer 4: Committing this because it'll also eliminate a trailing space.

> * The PreviousColumnName MUST not be Null
> * The PreviousColumnName MUST be of type String and be the name use in previous versions of the schema.
> * The PreviousColumnName MUST NOT be present on schema versions created after the rename.

Reviewer 4: ```suggestion
* PreviousColumnName MUST not be null.
* PreviousColumnName MUST be of type String.
* PreviousColumnName MUST be the name used in previous versions of the schema.
* PreviousColumnName MUST NOT be present in schema versions created after the rename.
```

## Metadata ID

PreviousColumnName

## Metadata Name

Previous Column Name

## Content constraints

| Constraint      | Value            |
|:----------------|:-----------------|
| Feature level   | Conditional      |
| Allows nulls    | False            |

> | Data type       | Boolean          |

Reviewer 3: ```suggestion
| Data type       | String           |
```

Author: This does not work. This doesn't indicate the previous name it would just provide a boolean which isn't useful.

Reviewer 4: @rileyjenk He is proposing to change it _from_ Boolean _to_ String.  Really just a consistency bugfix; you've declared it to be String in the normative requirements above:

> * PreviousColumnName MUST be the name used in previous versions of the schema.

@ljadvey, given that you gave this a 👍: what am I missing?
